hack/e2e/test.sh

```diff
@@ -0,0 +1,110 @@
+if ! kubectl get secret venafi-image-pull-secret -n venafi; then
+    venctl iam service-accounts registry create \
+           --no-prompts \
+           --owning-team "${VEN_OWNING_TEAM}" \
+           --name "venafi-kubernetes-agent-e2e-registry-${RANDOM}" \
+           --scopes enterprise-cert-manager,enterprise-venafi-issuer,enterprise-approver-policy \
+    | jq '{
+            "apiVersion": "v1",
+            "kind": "Secret",
+            "metadata": {
+              "name": "venafi-image-pull-secret"
+            },
+            "type": "kubernetes.io/dockerconfigjson",
+            "stringData": {
+              ".dockerconfigjson": {
+                "auths": {
+                  "\(.oci_registry)": {
+                    "username": .username,
+                    "password": .password
+                  }
+                }
+              } | tostring
+            }
+          }' \
+    | kubectl create -n venafi -f -
+fi
+
+# Cache the Service account credentials for venafi-kubernetes-agent in the cluster
+# but this Secret will not be mounted by the agent.
+kubectl create ns venafi-kubernetes-agent-e2e || true
+if ! kubectl get secret cached-venafi-agent-service-account -n venafi-kubernetes-agent-e2e; then
+    venctl iam service-account agent create \
+           --no-prompts \
+           --owning-team "${VEN_OWNING_TEAM}" \
+           --name "venafi-kubernetes-agent-e2e-agent-${RANDOM}" \
+    | jq '{
+            "apiVersion": "v1",
+            "kind": "Secret",
+            "metadata": {
+              "name": "cached-venafi-agent-service-account"
+            },
+            "stringData": {
+              "privatekey.pem": .private_key,
+              "client-id": .client_id
+            }
+          }' \
+    | kubectl create -n venafi-kubernetes-agent-e2e -f -
+fi
+
+export VENAFI_KUBERNETES_AGENT_CLIENT_ID="not-used-but-required-by-venctl"
+venctl components kubernetes apply \
+       --venafi-kubernetes-agent \
+       --venafi-kubernetes-agent-version "$VERSION" \
+       --venafi-kubernetes-agent-values-files "${script_dir}/values.venafi-kubernetes-agent.yaml" \
+       --venafi-kubernetes-agent-custom-image-registry "${OCI_BASE}/images" \
+       --venafi-kubernetes-agent-custom-chart-repository "oci://${OCI_BASE}/charts"
+
+privatekey=$(kubectl get secret cached-venafi-agent-service-account \
+                     --namespace venafi-kubernetes-agent-e2e \
+                     --template="{{index .data \"privatekey.pem\" | base64decode}}")
+clientid=$(kubectl get secret cached-venafi-agent-service-account \
+                   --namespace venafi-kubernetes-agent-e2e \
+                   --template="{{index .data \"client-id\" | base64decode}}")
+jwt=$(step crypto jwt sign \
+                   --key <(sed 's/ PRIVATE KEY/ EC PRIVATE KEY/g' <<<"$privatekey") \
+                   --aud api.venafi.cloud/v1/oauth/token/serviceaccount \
+                   --exp "$(date -d '+30 minutes' +'%s')" \
```

Comment:
nit: this doesn't work on macOS; this would work:

```suggestion
                   --exp "$([ "$(uname)" = "Darwin" ] && date -v +30M +'%s' || date -d '+30 minutes' +'%s')" \
```

Reply:
Done.